biz.aQute.bndlib/src/aQute/bnd/osgi/Builder.java

```diff
@@ -2086,4 +2127,146 @@ public String system(boolean allowFail, String command, String input) throws IOE
 		return cachedSystemCalls.computeIfAbsent(key, asFunction(k -> super.system(allowFail, command, input)));
 	}
 
+	private Function<Duplication, Optional<Resource>> parseDupStrategy(Map<String, String> extra) {
+		String onduplicate = extra.get(DUP_STRATEGY);
+		return Duplication.doDuplicate(onduplicate, this);
+	}
+
+
+	/**
+	 * Handles how duplicate resources are handled in -includeresource
+	 * instruction.
+	 */
+	public record Duplication(String path, Resource existing, Resource candidate) {
```

Comment:
Could we make it package private instead of public? Now it is public API and maybe we want to move it to Jar or so in the future, it seems potentially useful for other cases.

Reply:
Sure. will push in a minute